src/main/java/org/opensearch/neuralsearch/processor/rerank/ByFieldRerankProcessor.java

```diff
@@ -0,0 +1,280 @@
+
+        if (!validateSearchHits(searchHits, listener)) {
+            return;
+        }
+
+        List<Float> scores = new ArrayList<>(searchHits.length);
+
+        for (SearchHit hit : searchHits) {
+            Tuple<? extends MediaType, Map<String, Object>> mediaTypeAndSourceMapTuple = getMediaTypeAndSourceMapTuple(hit);
+            Map<String, Object> sourceAsMap = mediaTypeAndSourceMapTuple.v2();
+
+            Object val = getValueFromSource(sourceAsMap, targetField).get();
+            scores.add(((Number) val).floatValue());
+
+            if (keepPreviousScore) {
+                sourceAsMap.put("previous_score", hit.getScore());
+            }
+
+            if (removeTargetField) {
+                removeTargetFieldFromSource(sourceAsMap);
+            }
+
+            try {
+                XContentBuilder builder = XContentBuilder.builder(mediaTypeAndSourceMapTuple.v1().xContent());
+                builder.map(sourceAsMap);
+                hit.sourceRef(BytesReference.bytes(builder));
+            } catch (IOException e) {
+                listener.onFailure(new RuntimeException(e));
+                return;
+            }
+        }
+
+        listener.onResponse(scores);
+    }
+
+    /**
+     * This helper method is used to initialize the path to take to get to the targetField
+     * to remove. It is implemented recursively to delete empty maps as a result of removing the
+     * targetField
+     * <hr>
+     * <b>This method assumes that the path to the mapping exists as checked by {@link #validateSearchHits(SearchHit[], ActionListener)}</b>
+     * As such no error cehcking is done in the methods implementing this functionality
+     * @param sourceAsMap the map of maps that contains the <code>targetField</code>
+     */
+    private void removeTargetFieldFromSource(Map<String, Object> sourceAsMap) {
+        String[] keys = targetField.split("\\.");
+        exploreMapAndRemove(sourceAsMap, keys, 0);
+    }
+
+    /**
+     * This recursive method traces the path to targetField in a sliding window fashion. It does so
+     * by passing the parent map and the key (child) to get to the targetField (lastChild). Once it is found it will
+     * be deleted. The consequence of this, is having to delete all subsequent empty maps , this is
+     * accounted for by the last check to see that the mapping should be removed.
+     * <hr>
+     * <b>This method assumes that the path to the mapping exists as checked by {@link #validateSearchHits(SearchHit[], ActionListener)}</b>
+     * As such no error cehcking is done in the methods implementing this functionality
+     * @param sourceAsMap the map of maps that contains the <code>targetField</code>
+     * @param keys The keys used to traverse the nested map
+     * @param currentKeyIndex A sentinel to get the current key to look at
+     */
+    private void exploreMapAndRemove(Map<String, Object> sourceAsMap, String[] keys, int currentKeyIndex) {
```

Comment:
Understood the new commit [ce9bd7d](https://github.com/opensearch-project/neural-search/pull/932/commits/ce9bd7d7e5db8945ecc166aa8857f4bcfe15ad7f) addresses this feedback by making the method use a stack so that it can emulate the recursive solution iteratively!